test/suites-ostree/failed-upgrade.robot

```diff
@@ -0,0 +1,44 @@
+*** Settings ***
+Documentation       Tests related to upgrading MicroShift
+
+Resource            ../resources/common.resource
+Resource            ../resources/ostree.resource
+Library             Collections
+
+Suite Setup         Setup
+Suite Teardown      Teardown
+
+Test Tags           ostree
+
+
+*** Variables ***
+${USHIFT_HOST}      ${EMPTY}
+${USHIFT_USER}      ${EMPTY}
+
+${FAILING_REF}      ${EMPTY}
```

Comment:
Does the image referred to here need to already include something that will cause a failure, or does the TestAgent call on line 30 inject that?

Reply:
Image needs [microshift-test-agent RPM](https://github.com/openshift/microshift/tree/main/test/agent). We can create an alias: `ostree refs --create microshift/failing microshift/source`

Reply:
Since it is part of the test harness, we should add it to all of the blueprints. Can you do that in this PR?

If we need the alias, too, you could add that command to the image download script so it is always run, too. Do we want an alias for every image we build, or should we add them as we need them in a scenario?

Reply:
I added the alias to download. I only aliased microshift_source -> microshift_failing. I think we only need that one, so we can test that code under test behaves as expected during failure conditions